### PR TITLE
added css style classes property on prepared content table

### PR DESCRIPTION
**Pull request by Mir00r**

### CODE REVIEW

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/dtos/PreparedContentDto.kt [hunk 1]
```diff
@@ -26,6 +25,9 @@ class PreparedContentDto : BaseDto() {
     @ApiModelProperty(required = true)
     var placeholderValues: HashMap<String, String> = HashMap()
 
+    @JsonProperty("css_style_classes")
+    var cssStyleClasses: String? = null
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/models/dtos/PreparedContentDto.kt]: Should be @NotNull
default ""

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt [hunk 2]
```diff
@@ -33,6 +33,8 @@ class PreparedContent : BaseEntity() {
     @Column(name = "status", nullable = false)
     lateinit var status: ContentStatuses
 
+    @Column(name = "css_style_classes")
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt]: change it to `css_classes`
style is redundant

File: src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt [hunk 3]
```diff
@@ -66,7 +66,7 @@ class PreparedContentServiceBean @Autowired constructor(
         this.validate(entity)
         val placeholders = entity.placeholderValues.mapValues {
             if (it.key == DYNAMIC_CONTENT_KEY) {
-                HtmlTable.fromJson(it.value, listOf("dynamic-table"))
+                HtmlTable.fromJson(it.value, entity.cssStyleClasses?.split(" ") ?: listOf("dynamic-table"))
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt]: ` HtmlTable.fromJson(it.value, entity.cssStyleClasses?.split(" ") `

To be honest, we could just take a string as input in library, there's no need to take list. but for now it looks algiht.

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/dtos/PreparedContentDto.kt (hunk 1 above)
Comment (Mir00r) [reply src/main/kotlin/com/example/cms/domains/preparedcontents/models/dtos/PreparedContentDto.kt]: Done

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt (hunk 2 above)
Comment (Mir00r) [reply src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt]: Done

File: src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt (hunk 3 above)
Comment (Mir00r) [reply src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt]: @sayemkcn vai to be honest same thing me also thinking

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt [hunk 4]
```diff
@@ -33,6 +33,8 @@ class PreparedContent : BaseEntity() {
     @Column(name = "status", nullable = false)
     lateinit var status: ContentStatuses
 
+    @Column(name = "css_classes")
+    var cssClasses: String? = null
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/models/entities/PreparedContent.kt]: Entity is still nullable

File: src/main/kotlin/com/example/cms/domains/preparedcontents/models/mappers/PreparedContentMapper.kt [hunk 5]
```diff
@@ -33,6 +29,7 @@ class PreparedContentMapper @Autowired constructor(
                 this.placeholderValues[it.key] = it.value
             }
 
+            this.cssClasses = entity.cssClasses
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/models/mappers/PreparedContentMapper.kt:32]: have you tested it? can we assign null to a non nullable field?

File: src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt [hunk 6]
```diff
@@ -66,7 +66,7 @@ class PreparedContentServiceBean @Autowired constructor(
         this.validate(entity)
         val placeholders = entity.placeholderValues.mapValues {
             if (it.key == DYNAMIC_CONTENT_KEY) {
-                HtmlTable.fromJson(it.value, listOf("dynamic-table"))
+                HtmlTable.fromJson(it.value, entity.cssClasses?.split(" ") ?: listOf("dynamic-table"))
```
Comment (sayemoid) [inline src/main/kotlin/com/example/cms/domains/preparedcontents/services/beans/PreparedContentServiceBean.kt:69]: `entity.cssClasses?.split(" ") ?: listOf("dynamic-table")`

this field is non-null, so what's the point of `?: listOf("dynamic-table")`

**Pull request closed without merging**